sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1213,16 +1216,25 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
             ctx)
```

Comment:
I think this check should go first.

Reply:
oh, because `val schema = StructType(dataCols ++ partitionCols)` is defined, if this check goes first, it will shadow the next check `if (tableDesc.partitionColumnNames.nonEmpty)`.

Reply:
then can we check `dataCols` directly?

Reply:
Ok. That's good.